sig-autoscaling/charter.md

```diff
@@ -0,0 +1,74 @@
+# SIG Autoscaling Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
+the Roles and Organization Management outlined in [sig-governance].
+
+## Scope
+
+Covers development and maintenance of componets for automated scaling in
+Kubernetes.  This includes automated vertical and horizontal pod
+autoscaling, initial resource estimation, cluster-proportional system
+component autoscaling, and autoscaling of Kubernetes clusters themselves.
+
+### In scope
+
+- Autoscaling-related API objects, such as the HorizontalPodAutoscaler and
+  VerticalPodAutoscaler
+
+- Autoscaling-related tools, such as the cluster autoscaler and
+  cluster-proportional scaling tools (e.g. pod-nanny)
+
+- Ensuring API interfaces (the scale subresource) are availble and usable
+  to enable other SIG to write autoscalable objects, and enable people to
+  interact with those interfaces.
+
+[Link to SIG section in sigs.yaml][sigs.yaml]
+
+#### Code, Binaries and Services
+
+- Components and utilities that take automated action to scale a component
+  on the cluster
+
+- Components and utilities that take automated action to scale the cluster
+  itself
+
+- Parts of kubectl and client-go for interacting with autoscaling-related
+  API objects (e.g. HPA), as well as scaling interfaces (e.g. the
+  polymorphic scale client).
+
+#### Cross-cutting and Externally Facing Processes
+
+- Reviewing implementations of the scale subresource to ensure that
+  autoscaling behaves properly.
+
+- Coordinating with SIG Instrumentation to ensure that metrics APIs are
+  suitable for autoscaling on.
+
+### Out of scope
+
+- Testing performance at scale (this falls under the purview of [SIG
+  Scalability].
```

Comment:
We certainly do it for cluster autoscaler. And I don't think sig-scalability is doing any tests for HPA. Maybe we should reconsider? I would argue some level of scalability should be pre-requisite for HPAv2 GA, which would make it very much our problem.

Reply:
SIG scalability and SIG autoscaling are often confused in issue requests, and I kind-of wanted to make a clear deliniation.  It doesn't mean we can't test our *own* components, but we're not responsible for general cluster scale testing (e.g. if you can't launch 100000 pods at once, it's not our job to look into why).  I'll try to clarify.

Reply:
Ah, gotcha, I completely misunderstood what you meant.